### ISSUE TITLE
Correct handling for Cc field for single-commit submissions

### ISSUE THREAD

**Issue by newren** [gitgitgadget/gitgitgadget#220]
Quoting a conversation on the git mailing list (from https://lore.kernel.org/git/nycvar.QRO.7.76.6.2004041546290.46@tvgsbejvaqbjf.bet/):

> > > I'm not sure how cc is supposed to work with GGG, but it clearly didn't
> > > here. :)
> >
> > Yeah, I clearly don't either.  I even looked up another submission
> > from Dscho (https://github.com/git/git/pull/728) and attempted to
> > mimic it, but still managed to get it wrong somehow and I don't know
> > how.
> 
> It might be a bug in the way I implemented folding the cover letter into
> single-patch mails. I simply might not pick up the Cc:s there.
> 
> However, I am seriously under water right now and won't be able to work on
> a fix for that. Would you mind opening a ticket at
> https://github.com/gitgitgadget/gitgitgadget/issues/new?
> 
> Thanks,
> Dscho

**Comment by webstech** [comment]
> Quoting a conversation on the git mailing list (from https://lore.kernel.org/git/nycvar.QRO.7.76.6.2004041546290.46@tvgsbejvaqbjf.bet/):
> 
> > > I'm not sure how cc is supposed to work with GGG, but it clearly didn't
> > > here. :)

This came up in [git/git#740](https://github.com/git/git/pull/740#issuecomment-605523086) as well.  The cc is treated as the body when there is no body/description.

> > Yeah, I clearly don't either.  I even looked up another submission
> > from Dscho ([git/git#728](https://github.com/git/git/pull/728)) and attempted to

Markdown format on that one looks like it should be okay.

**Comment by dscho** [comment]
The code for handling the `Cc:` footers is [in the `parsePullRequestBody()` method](https://github.com/gitgitgadget/gitgitgadget/blob/c8e3d6b7b3fb0affcae800e2ffe2645cd7e9bb66/lib/patch-series.ts#L308-L316). The enclosing function is called [in `parsePullRequest()`](https://github.com/gitgitgadget/gitgitgadget/blob/c8e3d6b7b3fb0affcae800e2ffe2645cd7e9bb66/lib/patch-series.ts#L266), which in turn [is called from `getFromNotes()`](https://github.com/gitgitgadget/gitgitgadget/blob/c8e3d6b7b3fb0affcae800e2ffe2645cd7e9bb66/lib/patch-series.ts#L203), i.e. the pseudo-constructor of the `PatchSeries` class (a constructor cannot be `async`, therefore we have to resort to this static method).

Therefore, the existing code should have picked up the `Cc:` information _long_ before `generateAndSend()` [folds the cover letter into the single patch mail](https://github.com/gitgitgadget/gitgitgadget/blob/c8e3d6b7b3fb0affcae800e2ffe2645cd7e9bb66/lib/patch-series.ts#L804-L823).

Curious.

**Comment by dscho** [comment]
FWIW the gitgitgadget/gi PR that triggered this bug is here: https://github.com/git/git/pull/739

**Comment by dscho** [comment]
>  The cc is treated as the body when there is no body/description.

Ah, but of course. Yes, that's it.

**Comment by dscho** [comment]
The fix should be as simple as changing https://github.com/gitgitgadget/gitgitgadget/blob/c8e3d6b7b3fb0affcae800e2ffe2645cd7e9bb66/lib/patch-series.ts#L291 thusly:

```diff
-        const match = prBody.match(/^([^]+)\n\n([^]+)$/);
+        let match = prBody.match(/^([^]+)\n\n([^]+)$/);
+        if (!match && !prBody.match(/\n\n/)) {
+            // handle PR descriptions that have no body, just footers
+            match = prBody.match(/^()([A-Z-az]+: [^]+)$/);
+        }
```

But of course this would also need a regression test.

**Comment by webstech** [comment]
Although PR #214 allowed empty bodies, there is still another check that needs to be removed.
[`if (!prBody.length) {`](https://github.com/gitgitgadget/gitgitgadget/blob/c8e3d6b7b3fb0affcae800e2ffe2645cd7e9bb66/lib/patch-series.ts#L258).  Seems like this redundant check could be removed.

> match = prBody.match(/^()([A-Z-az]+: [^]+)$/);

`A-Za-z`.

> But of course this would also need a regression test.

Was already working with one.

**Comment by webstech** [comment]
PR #221 opened.  May need to add a test to ci-helper.test.ts to verify the empty description fix in #214 and now #221 does not have any more missed checks.

**Comment by dscho** [comment]
> > match = prBody.match(/^()([A-Z-az]+: [^]+)$/);
> 
> `A-Za-z`.

Or even `-A-Za-z`, as we also allow `Based-On:`.

> > But of course this would also need a regression test.
> 
> Was already working with one.

Thank you!!!
